addons/l10n_in_edi_ewaybill/models/account_move.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # Transaction Details
+    l10n_in_transaction_type = fields.Selection(
+        [
+            ("1", "Regular"),
+            ("2", "Bill To-Ship To"),
+            ("3", "Bill From-Dispatch From"),
+            ("4", "Combination of 2 and 3"),
+        ],
+        string="Transaction Type (E-waybill)",
+        tracking=True,
+    )
+    l10n_in_type_id = fields.Many2one("l10n.in.ewaybill.type", "Document Type", tracking=True)
+    l10n_in_subtype_id = fields.Many2one("l10n.in.ewaybill.type", "Sub Supply Type", tracking=True)
```

Comment:
@jva-odoo Could all of these fields be compute readonly False to have good default values?   Should we allow modifying these fields even if the invoice is posted?

Reply:
No need for default values.
and we can edit it after post because shipment can be created after an invoice is posted